woocommerce/google-listings-and-ads
# Display Free Listings and Paid Ads Campaigns in All Programs table in Dashboard

### Changes proposed in this Pull Request:

This PR displays the Free Listings and Paid Ads Campaigns in the All Programs table in the dashboard, and adds the navigation to the edit page for Free Listings and Paid Ads Campaign.

I have slightly changed the table columns. 

- Previously it was showing number of products, but we do not have that in our API and it is no longer in our latest figma. 
- I also added a new "Country" column which I think would be useful for users to select the right paid ads campaign to edit.
    - For Free Listings, it will display the first country name, and then followed with "+ X more" text.
- The "Spend" column is using `campaign.amount` data, which is the daily average cost entered by the users when creating the campaign.
- Free Listings will not display the red "Remove" button.

Note that the data in the Programs table is independent and not related to the date range filter located at the top of the dashboard page.

![Markup 2021-04-06 at 17 39 02](https://user-images.githubusercontent.com/65582817/113691516-13538680-96ff-11eb-8601-c0be4a7c2089.png)

This PR uses Ads Campaign API from PR #141.

### Outstanding item:

As per comments below, the All Programs table should display a "Status" column with "Eligible" as one of its possible values. This would be done later in issue https://github.com/woocommerce/google-listings-and-ads/issues/431.

### Screenshots:

Implementation:

![image](https://user-images.githubusercontent.com/417342/113734127-a86d7400-972d-11eb-8712-eac25248bcf1.png)

GIF to show navigation: 

![All Programs table and edit navigation 2](https://user-images.githubusercontent.com/417342/113738355-680ff500-9731-11eb-8268-465291e09916.gif)

### Detailed test instructions:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard
2. Click on the Edit button for Free Listings. It should navigate to the Edit Free Listing page.
3. Click on the Edit button for a Paid Ads Campaign. It should navigate to the Edit Paid Ads Campaign page.

### Changelog Note:

Displays the Free Listings and Paid Ads Campaigns in the All Programs table in the dashboard, and adds the navigation to the edit page for Free Listings and Paid Ads Campaign.
